textile/features.textile

```diff
@@ -161,6 +159,8 @@ h3(#restclient). RestClient
 ** @(RSC20d)@ All personally identifiable information, as much as is practicable, must be redacted or stripped completely before being submitted to Ably. Our intent is only to capture necessary information to debug issues in our own code
 ** @(RSC20e)@ Failures to log exceptions to the @errors.ably.io@ endpoint must be handled gracefully. This includes for example DNS failures, TCP/HTTP requests rejected, slow requests and internal failure errors. Additionally, as specified in @RSC20b2@, a failure to log an exception is logged with log level @info@ i.e. an exception reporting failure is not consider a client library @error@ or @warning@
 ** @(RSC20f)@ Any errors emitted by the library as a result of an internal failure must contain a status code @500@, an error code in the range @51000@ to @51999@ and a suitable error message. The error code must match one of "our common error codes":https://github.com/ably/ably-common/blob/main/protocol/errors.json
+* @(RSC22)@ @RestClient#batchPublish@ function takes a @BatchPublishSpec@ or an array of @BatchPublishSpec@s and sends then in a POST request to @/messages@ with the @newBatchResponse@ query param set to "true". When called with an array of @BatchPublishSpec@s it should return an array of @BatchResult<BatchPublishResult>@s. When called with a single @BatchPublishSpec@, the REST response will be an array containing a single @BatchResult<BatchPublishResult>@ which should be returned without the enclosing array.
```

Comment:
> When called with a single @BatchPublishSpec@, the REST response will be an array containing a single @BatchResult<BatchPublishResult>@ which should be returned without the enclosing array.

Should we consider making this optional, giving some sdks the option of not having an overload for the single-batchspec case? It's easy enough in things like typescript, but in eg golang there's no nice way of doing that, you'd have to just have a separate method, which seems not worth the faff

Reply:
Nice, thanks! updated